src/main/java/org/elasticsearch/app/api/server/IndexerController.java

```diff
@@ -86,9 +85,18 @@ public void stopIndex(@PathVariable String id) {
     }
 
     @PostMapping("/{source}/_clone/{target}")
-    public void cloneIndex(@PathVariable String source,@PathVariable String target){
-        configManager.cloneIndexes(source,target);
+    public void cloneIndex(@PathVariable String source, @PathVariable String target) {
+        configManager.cloneIndexes(source, target);
     }
 
+    @PostMapping("/import/configs")
+    public List<Integer> importConfigs(@RequestBody List<String> jsonConfig) {
+        return configManager.setAllConfigs(jsonConfig);
+    }
 
+    @DeleteMapping("/configs/{id}")
+    public void deleteIndex(@PathVariable String id, @RequestParam(required = false, defaultValue = "false") boolean deleteData) {
+        River river = configManager.getRiver(id);
```

Comment:
Why is it necessary to retrieve from DB the whole object in order to delete it in the next row? Would not em.getReference be enough?

Reply:
I am returning custom ConfigNotFindException that prevents Elasticsearch exception. With lazy load I used riverDAO.existsById(id) but it returned true even when the record does not exist. So I landed on this.

Reply:
I will try to update SpringBoot found some similar bug description I older versions.

Reply:
Changed to reference with getById()